Update async-await.md

uguratar opened the pull request:
issue nr: #16

Commit:
Update async-await.md

issue nr: #16

ozgunbal commented on docs/async-await.md:
```diff
@@ -1,9 +1,8 @@
-
-As a thought experiment imagine the following, a way to tell the JavaScript runtime to pause the executing of code on the `await` keyword when used on a promise and resume *only* once (and if) the promise returned from the function is settled.
+## Asenkron Bekleme
+Şöyle bir deney kurguladığımızı farzedelim, Javascript çalışma zamanında `await` anahtar kelimesi bir `promise`(söz) ile birlikte görüldüğü anda çalışmayı duraksayabiliyor ve sadece verilen sözün sonuca ulaşması tamamlandıktan sonra -sadece bir kerelik- çalışmaya kaldığı yerden devam ediyor.
```
farzedelim -> farz edelim. 
Daha türkçesi varsaymak var ayrıca :D 
Ek olarak sonuna 'etmek' gelen isimlerde yapısal değişiklik olmadığında ayrı yazılıyor. Bileşik olanına örnek olarak da seyir etmek değil, seyretmek.

Cümle bir yerden düşük gibi geldi. Benim önerim: "çalışmayı duraksayabiliyor" -> çalışması duraksayabilir
"devam ediyor" -> devam eder.

ozgunbal commented on docs/async-await.md:
```diff
@@ -13,36 +12,35 @@ async function foo() {
+* Fonksiyonun içerisinden *bir istisna fırlatabilme* kabiliyeti.
 
-This is exactly what generators allowed us to do! The thought experiment *is actually real* and is the `async`/`await` implementation in TypeScript / JavaScript. Under the covers it just uses generators.
+İşte bu saydıklarımız tam olarak *oluşturucular*ın yapmamıza olanak tanıdığı şey. Aslında farzedilen deney bir kurgu değil gerçeğin ta kendisidir ve  Javascript / Typescript dünyasında `asenkron`/`bekle` implementasyonu olarak ifade edilir. Perdenin arkasında oluşturucuları kullanır.
```
farzedilen -> farz edilen 
Öneri olarak: implementasyon için de uygulama / uygulanması kullanılabilir

uguratar replied:
http://implement.asyonturkcedegil.com/ =)

ozgunbal commented on docs/async-await.md:
````diff
@@ -13,36 +12,35 @@ async function foo() {
     }
 })
 ```
+`sozAlmakicinPaketle`'nin `oluşturucuyu` almak için sadece oluşturucu fonksiyonu çağırdığı bir durumda; daha sonra `generator.next()` ifadesi çağırılır, eğer dönen değer bir `söz` ise, `generator.next(result)` veya `generator.throw(error)` ifadelerinden biri duruma göre yakalanarak işlem sonlandırılır. 
````
then + catch'i eklesek daha iyi olabilir.
ifadelerden biri duruma göre "then veya catch" ile yakalanarak...

uguratar replied:
@ozgunbal çok teşekkürler, bu akşam hepsini gözden geçireyim, bi commit daha yapayım çözülsün bu iş. pinglerim hallolunca.

Commit:
Update async-await.md

Async Await çevirisi için çeşitli düzeltmeler, türkçeleştirmeler, düşük cümleleri adam etmeler falan.

uguratar commented:
@ozgunbal ping.
